### PR TITLE
2.296.2 Release notes

### LINKED ISSUE #2104: Environment variable "HOME=/github/home" now becomes "HOME" for self-hosted runner
**Describe the bug**
Previously (2.295.0), `docker create` [was executed](https://github.com/coiby/kexec-tools/runs/8007120416?check_suite_focus=true) with `-e "HOME=/github/home"` now it [is executed](https://github.com/coiby/kexec-tools/runs/8204074915?check_suite_focus=true) with `-e "HOME"`. It seems this is reason why [the job container failed to start](https://github.com/coiby/kexec-tools/runs/8204074915?check_suite_focus=true) as this is only difference between the jobs run by runner 2.295.0 and 2.296.1 respectively. 

**To Reproduce**
1. re-run [this job](https://github.com/coiby/kexec-tools/runs/8204074915?check_suite_focus=true) 

**Expected behavior**

The runner call `docker create` with `-e "HOME=/github/home"`
 
## Runner Version and Platform
Version of your runner?
2.296.1

OS of the machine running the runner? OSX/Windows/Linux/...
Linux

## What's not working?

Emulated Docker CLI using podman fails with the error "no such file or directory".

## Job Log Output
If applicable, include the relevant part of the job / step log output here.  All sensitive information should already be masked out, but please double-check before pasting here.

## Runner and Worker's Diagnostic Logs
If applicable, add relevant diagnostic log information.  Logs are located in the runner's `_diag` folder. The runner logs are prefixed with `Runner_` and the worker logs are prefixed with `Worker_`. Each job run correlates to a worker log.  All sensitive information should already be masked out, but please double-check before pasting here.

### PR DESCRIPTION
**Pull request by thboop**
Resolves #2104 and #2103

### CODE REVIEW

**Review by fhammerl — APPROVED** [review]
LGTM'

File: src/Runner.Worker/Container/DockerUtil.cs [hunk 1]
```diff
@@ -68,12 +68,37 @@ public static string CreateEscapedOption(string flag, string key)
...
+            // Dotnet escaping rules are weird here, we can only escape \ if it precedes a "
+            // If a double quotation mark follows two or an even number of backslashes, each proceeding backslash pair is replaced with one backslash and the double quotation mark is removed.
+            // If a double quotation mark follows an odd number of backslashes, including just one, each preceding pair is replaced with one backslash and the remaining backslash is removed; however, in this case the double quotation mark is not removed.
+            // https://docs.microsoft.com/en-us/dotnet/api/system.environment.getcommandlineargs?redirectedfrom=MSDN&view=net-6.0#remarks
+
+            // First, find any \ followed by a " and double the number of \ + 1.
+             value = Regex.Replace(value, @"(\\*)" + "\"", @"$1$1\" + "\"");
+            // Next, what if it ends in `\`, it would escape the end quote. So, we need to detect that at the end of the string and perform the same escape
+            // Luckily, we can just use the $ character with detects the end of string in regex
+            value = Regex.Replace(value, @"(\\+)$", @"$1$1");
```
Comment (TingluoHuang) [inline src/Runner.Worker/Container/DockerUtil.cs:99]: can we compile these regexes?

File: src/Runner.Worker/Container/DockerUtil.cs [hunk 2]
```diff
@@ -6,6 +6,9 @@ namespace GitHub.Runner.Worker.Container
 {
     public class DockerUtil
     {
+        private static readonly Regex QuoteEscape = new Regex(@"(\\*)" + "\"", RegexOptions.Compiled);
```
Comment (TingluoHuang) [inline src/Runner.Worker/Container/DockerUtil.cs:9]:
```suggestion
        private static readonly Regex QuoteEscape = new Regex(@"(\\*)\"", RegexOptions.Compiled);
```

**Review by TingluoHuang — APPROVED** [review]

File: src/Runner.Worker/Container/DockerUtil.cs (hunk 2 above)
Comment (TingluoHuang) [reply src/Runner.Worker/Container/DockerUtil.cs:9]: you can ignore me if the original way was for readability.

**Pull request merged**